thrust/examples/sum_columns.cu

```diff
@@ -0,0 +1,79 @@
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/random.h>
+#include <thrust/reduce.h>
+#include <thrust/transform.h>
+#include <thrust/universal_vector.h>
+
+#include <cuda/std/mdspan>
+
+#include <iomanip>
+#include <iostream>
+
+#include "include/host_device.h"
+
+int main()
+{
+  const int rows = 32;
+  const int cols = 16;
+
+  // Create a 2D multidimensional array of ints.
+  thrust::universal_vector<int> data(rows * cols, 42);
+  cuda::std::mdspan M(thrust::raw_pointer_cast(data.data()), rows, cols);
+
+  // Create a range to the flat linear index space for the multidimensional array.
+  auto flat_idx_begin = thrust::make_counting_iterator(0);
+  auto flat_idx_end   = flat_idx_begin + M.size();
+
+  // Fill the array with pseudorandom inputs in parallel on the device.
+  thrust::transform(thrust::device, flat_idx_begin, flat_idx_end, M.data_handle(), [] __host__ __device__(int flat) {
```

Comment:
Suggestion: This seems to transform a counting iterator. Let's use `thrust::tabulate` for that.